src/main/scala/com/azavea/modellab/Node.scala

```diff
@@ -15,96 +15,162 @@ import scala.collection.mutable
+) extends Node {
   require(n.extent <= 256, "Maximum Neighborhood extent is 256 cells")
   def buffer(bounds: GridBounds, cells: Int) = GridBounds(
-    math.max(0, bounds.colMin - cells), 
+    math.max(0, bounds.colMin - cells),
     math.max(0, bounds.rowMin - cells),
-    bounds.colMax + cells, 
+    bounds.colMax + cells,
     bounds.rowMax + cells)
 
   def calc(zoom: Int, bounds: GridBounds) = {
+    val rasterRDD = input(zoom, bounds)
+    val metaData = rasterRDD.metaData
     val extended = buffer(bounds, 1)
     println(s"FOCAL: $bounds -> $extended")
-    val rdd = layer(zoom, extended)
-    FocalOperation(rdd, Square(1), Some(bounds))(geotrellis.raster.op.focal.Min.apply)
-      .filter { case (key, _) => bounds.contains(key.col, key.row) } // filter buffer tiles, they contain no information      
+    val tileRDD = FocalOperation(rasterRDD, n, Some(bounds))(op)
+      .filter { case (key, _) => bounds.contains(key.col, key.row) } // filter buffer tiles, they contain no information
+    new RasterRDD(tileRDD, metaData)
   }
 }
 
-case class LocalAdd(a: Node, b: Node) extends Node {
+case class AspectOp(
+  op: (Tile, Neighborhood, Option[GridBounds], CellSize) => Tile,
+  input: Node,
+  n: Neighborhood
+) extends Node {
   def calc(zoom: Int, bounds: GridBounds) = {
-    a(zoom, bounds).join(b(zoom, bounds)).map { case (key, (tile1, tile2)) =>
-      key -> Add(tile1, tile2)
-    }
+    val rasterRDD = input(zoom, bounds)
+    val metaData = rasterRDD.metaData
+    val cs = metaData.layout.rasterExtent.cellSize
+    val tileRDD = rasterRDD map { case (key, tile) => key -> op(tile, n, None, cs) }
+    new RasterRDD(tileRDD, metaData)
   }
 }
 
-case class LocalSubtract(a: Node, b: Node) extends Node {
+case class SlopeOp(
+  op: (Tile, Neighborhood, Option[GridBounds], CellSize, Double) => Tile,
+  input: Node,
+  n: Neighborhood,
+  z: Double
+) extends Node {
   def calc(zoom: Int, bounds: GridBounds) = {
-    a(zoom, bounds).join(b(zoom, bounds)).map { case (key, (tile1, tile2)) =>
-      key -> Subtract(tile1, tile2)
-    }
+    val rasterRDD = input(zoom, bounds)
+    val metaData = rasterRDD.metaData
+    val cs = metaData.layout.rasterExtent.cellSize
+    val tileRDD = rasterRDD map { case (key, tile) => key -> op(tile, n, None, cs, z) }
+    new RasterRDD(tileRDD, metaData)
   }
 }
 
-case class LocalMultiply(a: Node, b: Node) extends Node {
+case class LocalUnaryOp(
+  op: Function1[Tile, Tile],
+  input: Node
+) extends Node {
   def calc(zoom: Int, bounds: GridBounds) = {
-    a(zoom, bounds).join(b(zoom, bounds)).map { case (key, (tile1, tile2)) =>
-      key -> Multiply(tile1, tile2)
-    }
+    val rasterRDD = input(zoom, bounds)
+    val metaData = rasterRDD.metaData
+    val tileRDD = rasterRDD map { case (key, tile) => key -> op(tile) }
+    new RasterRDD(tileRDD, metaData)
+  }
+}
+
+case class LocalBinaryOp(
+  op: LocalTileBinaryOp,
+  input: Seq[Node],
+  const: Option[Int] = None // Constant value
```

Comment:
This should be `Option[Double]` then on processing check `rdd.metaData.cellType.isFloatingPoint` and branch to call `RasterRDD.mapTiles` withe either `Int` or `Double` version of the local op.